src/main/logutils.ts

```diff
@@ -373,32 +376,132 @@ const determineArenaMatchResult = (line: LogLine): any[] => {
 }
 
 /**
- * Guard functions for now to simply allow us to ignore
- * Mythic Keystone encounters because we can't actually handle them
- * right now.
+ * Handle a log line for CHALLENGE_MODE_START
  */
-function handleChallengeModeStartLine (_line: LogLine): void {
-    isChallengeModeActive = true;
-}
+function handleChallengeModeStartLine (line: LogLine): void {
+    // It's impossible to start a keystone dungeon while another one is in progress
+    // so we'll just remove the existing one and make a new one when `CHALLENGE_MODE_START`
+    // is encountered.
+    if (activeChallengeMode) {
+        console.warn("[ChallengeMode] A Challenge Mode instance is already in progress; abandoning it.")
+    }
+    videoStartDate = line.date();
 
-function handleChallengeModeEndLine (_line: LogLine): void {
-    isChallengeModeActive = false;
-}
+    const zoneName = line.args[2];
+    const mapId = parseInt(line.args[3], 10);
+
+    if (!(mapId in dungeonTimersByMapId && mapId in dungeonsByMapId)) {
+        console.error(`[ChallengeMode] Invalid/unsupported mapId for Challenge Mode dungeon: ${mapId} ('${zoneName}')`)
+    }
+
+    const dungeonAffixes = line.args[5].map((v: string) => parseInt(v, 10));
+
+    activeChallengeMode = new ChallengeModeDungeon(
+        videoStartDate.getTime(),    // Start time
+        dungeonTimersByMapId[mapId], // Dungeon timers
+        parseInt(line.args[2], 10),  // zoneId
+        mapId,                       // mapId
+        parseInt(line.args[4], 10),  // Keystone Level
+        dungeonAffixes,              // Array of affixes, as numbers
+    )
+
+    activeChallengeMode.addVideoSegment(new ChallengeModeVideoSegment(
+        VideoSegmentType.Trash, videoStartDate, 0
```

Comment:
_Theater of Pain_ comes to mind as one such dungeon where it was a regular thing to skip the initial 3 mobs, and bring the 4th on top of the boss - besides that I don't recall any dungeons where doing two bosses without any trash is possible; HOWEVER even if that was the case there would probably be a skip strategy that would be relevant to have a beginning timestamp of.

I could label the first segment as 'Intro' segment instead, if you want?

Reply:
If it's just a case of labelling then I don't think I care too much -- If it's functionally a non-issue and the bottom line is we just occasionally label some strats wrongly then that seems like an easy fix to make when people complain about it rather than worry now.

Reply:
It's just labeling. I guess the "real fix" here is to make it so that if a timeline segment (video segment) is too short to have value, e.g. 10 seconds, then it'll be discarded and whatever comes next (like a boss) will take its place, but I think those refinements are better to keep for future improvements.

Reply:
Awesome - totally happy with that.